core/services/pipeline/task.lessthan_test.go

```diff
@@ -0,0 +1,142 @@
+		{"string, lt large value", "1.23", "1000000000000000000", true},
+		{"large string, lt large value", "10000000000000000001", "1000000000000000000", false},
+
+		{"int, true", int(2), "100", true},
+		{"int, false", int(2), "-5", false},
+
+		{"int8, true", int8(2), "100", true},
+		{"int8, false", int8(2), "-5", false},
+
+		{"int16, true", int16(2), "100", true},
+		{"int16, false", int16(2), "-5", false},
+
+		{"int32,true", int32(2), "100", true},
+		{"int32, false", int32(2), "-5", false},
+
+		{"int64, true", int64(2), "100", true},
+		{"int64, false", int64(2), "-5", false},
+
+		{"uint, true", uint(2), "100", true},
+		{"uint, false", uint(2), "-5", false},
+
+		{"uint8, true", uint8(2), "100", true},
+		{"uint8, false", uint8(2), "-5", false},
+
+		{"uint16, true", uint16(2), "100", true},
+		{"uint16, false", uint16(2), "-5", false},
+
+		{"uint32, true", uint32(2), "100", true},
+		{"uint32, false", uint32(2), "-5", false},
+
+		{"uint64, true", uint64(2), "100", true},
+		{"uint64, false", uint64(2), "-5", false},
+
+		{"float32, true", float32(1.23), "10", true},
+		{"float32, false", float32(1.23), "-5", false},
+
+		{"float64, true", float64(1.23), "10", true},
+		{"float64, false", float64(1.23), "-5", false},
+	}
+
+	for _, test := range tests {
+		assertOK := func(result pipeline.Result, runInfo pipeline.RunInfo) {
+			assert.False(t, runInfo.IsPending)
+			assert.False(t, runInfo.IsRetryable)
+			require.NoError(t, result.Error)
+			require.Equal(t, test.want, result.Value.(bool))
+		}
+		t.Run(test.name, func(t *testing.T) {
+			t.Run("without vars through job DAG", func(t *testing.T) {
+				vars := pipeline.NewVarsFrom(nil)
+				task := pipeline.LessThanTask{BaseTask: pipeline.NewBaseTask(0, "task", nil, nil, 0), Right: test.right}
+				assertOK(task.Run(testutils.Context(t), logger.TestLogger(t), vars, []pipeline.Result{{Value: test.left}}))
+			})
+			t.Run("without vars through input param", func(t *testing.T) {
+				vars := pipeline.NewVarsFrom(nil)
+				task := pipeline.LessThanTask{
+					BaseTask: pipeline.NewBaseTask(0, "task", nil, nil, 0),
+					Left:     fmt.Sprintf("%v", test.left),
+					Right:    test.right,
+				}
+				assertOK(task.Run(testutils.Context(t), logger.TestLogger(t), vars, []pipeline.Result{}))
+			})
+			t.Run("with vars", func(t *testing.T) {
+				vars := pipeline.NewVarsFrom(map[string]interface{}{
+					"foo":   map[string]interface{}{"bar": test.left},
+					"chain": map[string]interface{}{"link": test.right},
+				})
+				task := pipeline.LessThanTask{
+					BaseTask: pipeline.NewBaseTask(0, "task", nil, nil, 0),
+					Left:     "$(foo.bar)",
+					Right:    "$(chain.link)",
+				}
+				assertOK(task.Run(testutils.Context(t), logger.TestLogger(t), vars, []pipeline.Result{}))
+			})
+		})
+	}
+}
+
+func TestLessThanTask_Unhappy(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name              string
+		right             string
+		left              string
```

Comment:
nit/ left before right?
```suggestion
		left              string
		right             string
```